api/pkg/backend/service.go

```diff
@@ -770,27 +699,85 @@ func (s *APIService) DeleteTier(request *restful.Request, response *restful.Resp
 	id := request.PathParameter("id")
 	log.Infof("Received request for deleting tier: %s\n", id)
 	ctx := common.InitCtxWithAuthInfo(request)
+
 	res, err := s.backendClient.GetTier(ctx, &backend.GetTierRequest{Id: id})
 	if err != nil {
 		log.Errorf("failed to get tier details: %v\n", err)
 		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
-	// check whether tier is empty
-	if len(res.Tier.Backends) != 0 {
-		log.Errorf("failed to delete tier because tier is not empty has backends: %v\n", err)
-		response.WriteError(http.StatusInternalServerError, err)
-		return
+
+	//check whether backends of tier has no buckets
+	for _, backendId := range res.Tier.Backends {
+		count, err := s.GetBucketsCount(backendId, ctx)
+		if err != nil {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		}
+		if count != 0 {
+			errMsg := fmt.Sprintf("failed to delete tier because backendId %v has buckets(not empty)", backendId)
+			log.Error(errMsg)
+			response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+			return
+		}
 	}
 
 	_, err = s.backendClient.DeleteTier(ctx, &backend.DeleteTierRequest{Id: id})
 	if err != nil {
 		log.Errorf("failed to delete tier: %v\n", err)
-		err1 := errors.New("failed to delete tier because tier has associated backends")
+		err1 := errors.New("failed to delete tier")
 		response.WriteError(http.StatusInternalServerError, err1)
 		return
 	}
 	log.Info("Delete tier  successfully.")
 	response.WriteHeader(http.StatusOK)
 	return
 }
+
+//checking whether backend has buckets
+func (s *APIService) GetBucketsCount(backendId string, ctx context.Context) (int, error) {
+	res2, err := s.s3Client.ListBuckets(ctx, &s3.BaseRequest{})
+	if err != nil {
+		return -1, err
+	}
+
+	count := 0
+	res1, err1 := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+	if err1 != nil {
+		log.Errorf("failed to get backend details: %v\n", err)
+		return -1, err1
+	}
+
+	for _, v := range res2.Buckets {
+		backendname := res1.Backend.Name
+		if backendname == v.DefaultLocation {
+			count++
+		}
+	}
+	return count, err
+}
+
+//give the list of invalid backends
+func (s *APIService) GetInvalidBackends(objects []string, ctx context.Context) ([]string, error) {
```

Comment:
https://github.com/sodafoundation/multi-cloud/pull/1273 has IsInvalidBackedns() function too..check and keep one